app/data/movielens-api.js

```diff
@@ -0,0 +1,29 @@
+const movielens = require("movielens");
+const validators = require("../utils/validators");
+
+let cookie;
+
+async function movieLensLogin() {
+  cookie = await movielens.login(
+    process.env.MOVIELENS_USERNAME,
+    process.env.MOVIELENS_PASSWORD
+  );
+  console.log(`Logged in to Movielens successfully. Cookie: ${cookie}`);
+}
+
+movieLensLogin();
```

Comment:
Everything looks good! Just one thing we need to be mindful of is that if this gets deployed to Heroku this will fetch a cookie when the server is first started. It's possible that the cookie expires after a day or so, and the only way to refresh our cookie would be to restart the server. We may want to consider logging in and getting a cookie every time a request is made. What do you think?

Reply:
Good point. From the cookie itself, it seems that the expiration date is one year later:
`Logged in to Movielens successfully. Cookie: ml4_session=142e7ad13d16f09065fa60998414ac7d41ab8e2e_2b5ab180-2676-4fc1-a079-66f548fd4764; Max-Age=31536000; Expires=Tue, 09 Aug 2022 00:09:33 GMT; SameSite=Lax; Path=/; Secure; HTTPOnly`

But I'm not sure if that is actually the case. We can test it and see if it expires that often. Ideally, we need to discover if it was expired and refresh it. I don't think logging in each time we make a request is a good idea. In the best case, that adds one additional request to Movielens each time.